iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp

```diff
@@ -149,13 +151,24 @@ class OutlineDispatchRegionsPass
 
   void runOnOperation() override {
     // Convert each dispatch region into a flow.executable + dispatch op.
-    for (auto funcOp : getOperation().getOps<mlir::FuncOp>()) {
+    for (auto it : llvm::enumerate(getOperation().getOps())) {
+      Operation &op = it.value();
+      if (!op.hasTrait<OpTrait::FunctionLike>()) continue;
+
+      // Generate a nice name if possible.
+      std::string opName;
+      if (auto funcOp = llvm::dyn_cast<mlir::FuncOp>(op)) {
+        opName = funcOp.getName().str();
+      } else {
+        opName = std::string("_function_like_") + std::to_string(it.index());
```

Comment:
would be nice to have a case for initializers in here, given that they're the only other one we have now and seeing `_function_like_` everywhere would make me sad (`_initializer_` is probably good enough).

Reply:
Sold.